### PR TITLE
Workaround vscode bug with returning defaultBehavior from prepareRename

### PR DESCRIPTION
**Pull request by dibarbet**
VSCode bug here - https://github.com/microsoft/vscode-languageserver-node/issues/1363

Returning an explicit range instead of defaultBehavior works.

Resolves https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1916980

### CODE REVIEW

File: src/Features/LanguageServer/Protocol/Handler/Rename/PrepareRenameHandler.cs [hunk 1]
```diff
@@ -37,9 +37,7 @@ public TextDocumentIdentifier GetTextDocumentIdentifier(PrepareRenameParams requ
         if (symbolicRenameInfo.IsError)
             return null;
 
-        return new DefaultBehaviorPrepareRename
-        {
-            DefaultBehavior = true,
-        };
+        var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+        return ProtocolConversions.TextSpanToRange(symbolicRenameInfo.TriggerToken.Span, text);
```
Comment (dibarbet) [inline src/Features/LanguageServer/Protocol/Handler/Rename/PrepareRenameHandler.cs:41]: The range here is the initial range to replace - the client takes this range to prefill the rename dialog with the identifier to rename.

**Comment by davidwengier** [comment]
lol, so neither VS nor VS Code handle that response type? So glad we have a spec 😛

File: src/Features/LanguageServer/Protocol/Handler/Rename/PrepareRenameHandler.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Features/LanguageServer/Protocol/Handler/Rename/PrepareRenameHandler.cs:41]: Can you test that this works properly with things like `@foo` and we don't get weirdness.

Comment (dibarbet) [reply src/Features/LanguageServer/Protocol/Handler/Rename/PrepareRenameHandler.cs:41]: Added a test.  This also appears to work better than before as well.  For example, what happened before this change is
image

But now it looks like 
image

**Pull request merged**